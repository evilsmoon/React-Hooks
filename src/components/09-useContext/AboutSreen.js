import React, { useCallback, useContext } from 'react'
import { UserContext } from './UserContext'

// import './styles.css'

export const AboutSreen = () => {
    
    const {user , setUser} = useContext(UserContext);
    
    const handelLogout = ()=>{

        setUser({})
    }
    return (
        <div>
            <h1>AboutSreen</h1>
            <hr />
            <pre>
                {
                    JSON.stringify(user,null,3)
                }
            </pre>

            <button
                className='btn btn-secondary'
                onClick={handelLogout}
           >
                Logout
            </button>

        </div>
    )
}
