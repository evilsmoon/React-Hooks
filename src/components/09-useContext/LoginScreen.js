import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
    
    const { setUser} = useContext(UserContext)
    
    // console.log(user ,setUser)
    
    const user = {
        id: 123,
        name:'Gabriel'
    }
    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />
            <button
                className='btn btn-info'
                onClick={()=>{
                    setUser(user)
                }}
            >
                Login
           
            </button>
        </div>

    )
}
