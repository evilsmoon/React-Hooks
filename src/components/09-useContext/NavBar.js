import React from 'react'
import {  NavLink } from "react-router-dom";
export const NavBar = () => {
    return (

        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
            <NavLink className="navbar-brand" to="./">Use Context</NavLink>
            <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    <NavLink  exact activeClassName='active' className="nav-link nav-item" to="./">
                        Home
                    </NavLink>
                    <NavLink  exact activeClassName='active' className="nav-link nav-item" to="./about">
                        About
                    </NavLink>
                    <NavLink  exact activeClassName='active' className="nav-link nav-item" to="./login">
                        Login
                    </NavLink>

                </div>
            </div>
        </nav>
    )
}

