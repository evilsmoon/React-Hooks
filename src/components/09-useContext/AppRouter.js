import React from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
  } from "react-router-dom";
import { AboutSreen } from './AboutSreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';
import { NavBar } from './NavBar';
export const AppRouter = () => {
    return (
           <Router>
               <div>
                   <NavBar />
                   <div className='container'>

                   <Switch>
                       <Route exact path='/' component={ HomeScreen }/>
                       <Route exact path='/about' component={ AboutSreen }/>
                       <Route exact path='/login' component={ LoginScreen }/>
                       <Redirect to="/" />
                   </Switch>
                   </div>
               </div>
           </Router> 
    )
}
