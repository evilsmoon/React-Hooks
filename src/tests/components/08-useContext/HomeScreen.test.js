import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import React from 'react';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';



describe('test HomeScreen.js', () => {


    const user = {
        name: 'Gabriel',
        email: 'gmorale@gmail.com'

    }
    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen />
        </UserContext.Provider>
    )

    test('debe de mostrar correctamente <HomeScreen.js/>', () => {

        expect(wrapper).toMatchSnapshot();
    })

})

