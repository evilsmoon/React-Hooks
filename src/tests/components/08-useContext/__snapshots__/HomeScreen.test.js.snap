// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test HomeScreen.js debe de mostrar correctamente <HomeScreen.js/> 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Gabriel",
   "email": "gmorale@gmail.com"
}
  </pre>
</div>
`;
